custom_components/tuya_local/devices/nine_ev_charger.yaml

```diff
@@ -0,0 +1,262 @@
+name: EV charger
+products:
+  - id: bktb3jskdic1ar2t
+    name: Nine 32A EV charger
+primary_entity:
+  entity: sensor
+  name: Charger State
+  icon: "mdi:ev-station"
+  class: enum
+  dps:
+    - id: 3  # work_state
+      type: string
+      name: sensor
+      mapping:
+        - dps_val: charger_charging
+          value: Charging
+        - dps_val: charger_free
+          value: Available
+        - dps_val: charger_insert
+          value: Plugged in
+        - dps_val: charger_free_fault
+          value: Fault
+        - dps_val: charger_wait
+          value: Plugged in, not charging
+        - dps_val: charger_pause
+          value: Paused
+        - dps_val: charger_end
+          value: Charge finished
+        - dps_val: charger_fault
+          value: Fault (plugged in)
+    - id: 23  # system_version
+      type: string
+      name: system_version
+secondary_entities:
+  - entity: number
+    name: Charging current
+    category: config
+    class: current
+    icon: "mdi:ev-plug-type2"
+    dps:
+      - id: 4  # charge_cur_set
+        type: integer
+        name: value
+        unit: A
+        optional: true
+        range:
+          min: 6
+          max: 80
+        mapping:
+          - scale: 1
+  - entity: sensor
+    class: power
+    category: diagnostic
+    dps:
+      - id: 9
+        type: integer
+        optional: true
+        name: sensor
+        unit: kW
+        class: measurement
+        mapping:
+          - scale: 1000
+  - entity: binary_sensor
+    class: problem
+    category: diagnostic
+    dps:
+      - id: 10  # fault
+        type: bitfield
+        name: sensor
+        optional: true
+        mapping:
+          - dps_val: 0
+            value: false
+          - dps_val: 1
+            value: "Ov2 Cr Fault"
```

Comment:
binary_sensor mappings should be to booleans.
I suggest adding 2 extra attributes: fault_code with the raw integer code, and description with these mapped descriptions. See the recently added Cecotec config https://github.com/make-all/tuya-local/pull/2222/files#diff-8ca651a12d089c6826bf3817b0f031d1edd4938110267451db6d877c6bfab795R90

Reply:
Will do, thanks